src/AppInitializer.tsx

```diff
@@ -117,6 +119,20 @@ function AppInitializer(props: { children?: React.ReactNode }) {
     } else if (localStorageAppId) {
       setAppId(localStorageAppId)
     }
+
+    const searchParamRedirectURL = getSearchParam('redirectURL')
+    const localStorageRedirectURL = localStorage.getItem('sourceAppRedirectURL')
+    if (searchParamRedirectURL) {
```

Comment:
Look for "redirectURL" query parameter.  Only accept it if the hostname ends with .synapse.org for now (to lock it down).  I think this will work for all known use cases today.  MTB will send to the Synapse oauth signin app (hosted on a .synapse.org subdomain)

Reply:
what would mtb link look like? Something like this?  https://signin.synapse.org/?redirectUrl=https://staging.studies.mobiletoolbox.org

Reply:
Yes, exactly.  Where the full redirectURL (for SageAccountWeb) would be URL encoded.
So your appId would equal "MTB", and your redirectURL would equal "https://signin.synapse.org/?response_type=code&client_id=100123&scope=openid&state=1gn12e487&redirect_uri=https%3A%2F%2Fstudies.mobiletoolbox.org&claims=%7B%22id_token%22%3A%7B%22userid%22%3Anull%7D%7D"